Sources/MongoSwift/MongoCollection+Indexes.swift

```diff
@@ -60,6 +60,10 @@ public struct IndexOptions: Codable {
     /// is sent and the default collation of the collection server-side is used.
     public var collation: BSONDocument?
 
+    /// An arbitrary BSON type to help trace the index through
+    /// the database profiler, currentOp and logs. The default is to not send a value.
+    public var comment: BSON?
```

Comment:
These options are used to configure the index itself, whereas the "comment" fields we're adding in this PR are used to track operations. For that reason, the [Index Management spec](https://github.com/mongodb/specifications/blob/master/source/index-management.rst#standard-api) includes this option under `CreateIndexesOptions` (which are options for the `createIndex` operation), not `IndexOptions`.

Reply:
removed comment for `IndexOptions`, left in for `CreateIndexOptions`